### DISCUSSION TITLE
Difference between `--no-api` and `--hide-api`?

### DISCUSSION THREAD

**Discussion by thomthom** [lsegal/yard#1574] [Q&A] [answered]
What is the difference between using `--no-api` and `--hide-api`?

**Comment by lsegal** [comment, marked as answer]
1. `--no-api` does not take a TAG argument (optparse makes this hard to show) and will ignore objects that _do not have_ an `@api` tag attached. It's equivalent to `--api every --api other --api api --api used`. See [specs](https://github.com/lsegal/yard/blob/84c983da9157ab7a6eccbc7a1740f2e22c05b679/spec/cli/yardoc_spec.rb#L365) for examples.
2. `--hide-api TAG` will hide objects for a particular API tag.

Unfortunately using the `--[no-]` prefix in optparse does not allow you to take an argument afterward, hence the use of `--hide-api` for that behavior.
